src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs

```diff
@@ -296,9 +315,30 @@ private void MaybeSetCache(GetResponse getResponse, GetRequest command)
             if (changeVector == null)
                 return;
 
-            _cache.Set(cacheKey, changeVector, result);
+            _httpCache.Set(cacheKey, changeVector, result);
         }
 
         public override bool IsReadRequest => false;
+
+        public void Dispose() => _cached?.Dispose();
+
+        private class Cached : IDisposable
+        {
+            public readonly (HttpCache.ReleaseCacheItem Release, BlittableJsonReaderObject Cached)[] Values;
+
+            public Cached(int size)
+            {
+                Values = ArrayPool<(HttpCache.ReleaseCacheItem, BlittableJsonReaderObject)>.Shared.Rent(size);
+            }
+
+            public void Dispose()
+            {
+                for (int i = 0; i < Values.Length; i++)
```

Comment:
Values.Length might be much greater than requested size. Won't you get NRE here?

Reply:
Basically, it is struct so the dispose in `HttpCache.ReleaseCacheItem` will do nothing since its item is null.
But it is more appropriate to use the initialize size so I will change it